tests/test_viz.py

```diff
@@ -254,6 +254,80 @@ def test_plot_heatmap(ocx, api_data):
     assert all(chart.data.groupby("tax_id").max()["Relative Abundance"] > 0.01)
 
 
+def test_plot_heatmap_plots_all_nan_samples_with_nans(ocx, api_data):
+    import numpy as np
+
+    samples = ocx.Samples.where(project="4b53797444f846c4")
+
+    # Set abundances for first sample to be all NaN
+    all_nan_sample = samples[0]
+    samples._results.loc[all_nan_sample.primary_classification.id] = np.nan
+    assert len(samples.all_nan_classification_ids) == 1
+
+    all_nan_classification_id = samples.all_nan_classification_ids[0]
+    chart = samples.plot_heatmap(
+        top_n=10, title="my title", xlabel="my xlabel", ylabel="my ylabel", return_chart=True
+    )
+    chart_data = chart["data"]
+
+    # Samples with no abundances calculated for them should have all-NaN rows in chart data df
+    # Other classification_id rows should not have any `NaN`s
+    assert (
+        chart_data[(chart_data == all_nan_classification_id).any(axis=1)]["Relative Abundance"]
+        .isnull()
+        .all()
+        == True  # noqa
+    )
+    assert (
+        chart_data[(chart_data == samples[1].primary_classification.id).any(axis=1)][
+            "Relative Abundance"
+        ]
+        .isnull()
+        .any()
+        == False  # noqa
+    )
+
+    # Sample with all NaNs should be at the end
+    assert chart.encoding.x.sort[-1] == all_nan_sample.filename
+
+
+def test_plot_heatmap_on_df_plots_all_nan_samples_with_nans(ocx, api_data):
```

Comment:
what's the difference between this test and the one above it?

Reply:
This tests that the functionality works for `plot_heatmap` called on a `OneCodexAccesor` , ie `samples.to_df().ocx.plot_heatmap`. Since `OneCodexAccessor` uses its own version of `_results`, we need to make sure that we're getting the same functionality (by storing `all_nan_classification_ids` on both `SampleCollection`s and `OneCodexAccessor`s) in both cases